arch/arc/include/kernel_arch_data.h

```diff
@@ -36,9 +32,11 @@ extern "C" {
 #include <zephyr/types.h>
 #include <sys/util.h>
 #include <sys/dlist.h>
+
+#ifdef __cplusplus
+extern "C" {
 #endif
 
-#ifndef _ASMLANGUAGE
```

Comment:
other _ASMLANGUAGE checks in this commit have been moved, not removed. Are you sure this one can be removed?
The corresponding #endif in line 173/175 isn't removed.

Reply:
What's actually happening here is that the `#endif` on line 38 used to be the close of the `#ifndef _ASMLANGUAGE` up on line 33 that enclosed only the `#include` material, immediately followed by another `#ifndef _ASMLANGUAGE` for the structure definition.    So the previous code had two adjacent `#ifndef _ASMLANGUAGE/#endif` blocks, which I merged by removing the `#ifdef/#endif` pair.   I then inserted an `#ifdef __cplusplus/#endif` block which restored the `#endif` on 39.  The whole `#ifndef _ASMLANGUAGE` block is still closed down on line 173.

See https://github.com/zephyrproject-rtos/zephyr/blob/0409dd1ce83501b2e091edb9ccbf70e92e89a967/arch/arc/include/kernel_arch_data.h#L33 for the before state.

OK?

Reply:
thanks for clarifying, missed that when only looking at the changes and not complete file.